### ISSUE TITLE
S3 Error when computing diffs

### ISSUE THREAD

**Issue by dharrigan** [liquidz/antq#133]
Hi,

I think that doing diffs on s3 artifacts should be skipped, otherwise the following errors occur:

When resolving `S3://` artifacts for doing diffs, `antq` calls to the [io/reader](https://github.com/liquidz/antq/blob/de79d34bc4d951480f41cbb2e736cdc215a30d68/src/antq/util/maven.clj#L118) to return a stream. 

Unfortunately, the [java.io/reader](https://github.com/clojure/clojure/blob/5451cee06b9e31513a19e596e4e155d1f08d2a8d/src/clj/clojure/java/io.clj#L251) which extends the protocol to String, only is aware of [standard](https://github.com/clojure/clojure/blob/5451cee06b9e31513a19e596e4e155d1f08d2a8d/src/clj/clojure/java/io.clj#L236) `URL` schemas, such as `http`, `https` (and falls back to `file`). It doesn't know how to handle `s3://` URLs.

A MalformedUrlException is [thrown](https://github.com/clojure/clojure/blob/5451cee06b9e31513a19e596e4e155d1f08d2a8d/src/clj/clojure/java/io.clj#L257) and an attempt is then made to retrieve the artifact using a file based approach.

This causes the following errors to appear on the console:

```
Fetching pom from s3://maven.foo.com/releases/com/foo/bar/0.3.71/bar-0.3.71.pom failed because of the following error: s3:/maven.foo.com/releases/com/foo/bar/0.3.71/bar-0.3.71.pom (No such file or directory)
Fetching pom from s3://maven.foo.com/releases/com/foo/bar/0.3.71/bar-0.3.71.pom failed because of the following error: s3:/maven.foo.com/releases/com/foo/bar/0.3.71/bar-0.3.71.pom (No such file or directory)
```

(interestingly, notice the `s3:/` as the schema, I think that's because of the attempt to resolve it as a file).

I'm not 100% sure how to go about approaching this one. It *is* possible to register a new S3 `URLStreamHandler` when creating an instance of the URL, so this [line](https://github.com/clojure/clojure/blob/5451cee06b9e31513a19e596e4e155d1f08d2a8d/src/clj/clojure/java/io.clj#L256) would need to change to something like `(URL. nil x s3-schema-handler)` (see the constructor in java.net.URL)  (of course, the entire [extend](https://github.com/clojure/clojure/blob/5451cee06b9e31513a19e596e4e155d1f08d2a8d/src/clj/clojure/java/io.clj#L251) would need to be substituted in antq code).

-=david=-

**Comment by liquidz** [comment]
@dharrigan Just released v1.3.2.
Thank you so much!

**Comment by dharrigan** [comment]
You're most welcome! Happy to help where I can :-)
